### PR TITLE
Add onError function on service error

### PR DESCRIPTION
**Pull request by Akrion**
Adds an onError function which can be passed to monitor an exception being thrown by the search service

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR has lint errors. Please fix these and commit them.
      
<details>
  <summary>index.js</summary>
  <p><a href="https://github.com/smartprocure/contexture-client/blob/feature/HandleServiceError/s" target="_blank">s</a></p>
  
  <strong>'lastUpdateTime' is defined but never used.</strong>
  <code>Line 405: let service = sinon.spy(async (dto, lastUpdateTime) => { throw('service error') })</code>

  <strong>'e' is defined but never used.</strong>
  <code>Line 409: let onError = e => spy()</code>
</details>
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Could not run `duti:fix` command successfully
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c1c90aaac015025cd318c51381c6b8dcc756aa03
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
-﻿# 2.20.0
+﻿# 2.20.1
```
Comment (daedalus28) [inline CHANGELOG.md]: this should probably be a feature revision (minor release) since its a new thing

Comment (Akrion) [reply CHANGELOG.md]: There were conflicts ... jest resolved them

Comment (daedalus28) [reply CHANGELOG.md]: it should still be a minor and not patch revision :)

Comment (daedalus28) [reply CHANGELOG.md]: also, there's a lint error in your test case

**Pull request merged**